packages/bootstrap-4/test/descriptionField.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import DescriptionField from "../src/DescriptionField";
+import { JSONSchema7 } from "json-schema";
+import renderer from "react-test-renderer";
+
+describe("DescriptionField", () => {
+  const schema: JSONSchema7 = {
+    type: "array",
+    items: {
+      type: "string",
+    },
+  };
+  const mockHandler = (): void => void 0;
+  test("should return null when no description as a props is passed", () => {
+    const tree = renderer
+      .create(
+        <DescriptionField
+          uiSchema={{}}
+          idSchema={{
```

Comment:
Typedef for `idSchema` is quite complex for me to mock it. 

Currently it is wrong, it is throwing type error 

![image](https://user-images.githubusercontent.com/26347874/84891079-f17a0c00-b0b8-11ea-8c98-9a37050f6d7d.png)

Would appreciate some help here.

Typedef for `idSchema`

```ts
    export type FieldId = {
        $id: string;
    };

    export type IdSchema<T = any> = {
        [key in keyof T]: IdSchema<T[key]>;
    } &
        FieldId;

```

https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L124

Reply:
@anikethsaha why are we testing passing all these props to the component, since the component doesn't do anything with them? No matter what props you pass to it, it will always just return
```jsx
<h2 className="mt-5">{description}</h2>
```

Personally, I would only test the case where description is null, and one where it's a string.

Reply:
It will throw type error as all properties are marked as required https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L120

Reply:
and in `tsdx test` command, there is no way to bypass type checking

Reply:
@anikethsaha I see! @Xtremilicious maybe this is good input for deciding how to implement the type interfaces? If the type system is forcing to pass [all these props](https://github.com/rjsf-team/react-jsonschema-form/blob/b4824b9f44b84fced7187bc752d8614a3c90b72c/packages/core/index.d.ts#L120-L142) to a component that doesn't even use it, maybe that means that the type design is incorrect?

To be honest, I don't really understand how the material-ui theme gets away with this. If they do TypeScript checking, they should be running into the same issue in their tests, but also any TypeScript user of the library should be hitting the same error in runtime?

(Update: I just went to see how they handle it, and they... don't. They don't have tests for component types.)

To me this is saying that the path of least resistance is to tell the type system to expect only the appropriate props.

Reply:
Actually material ui theme is not testing all the components. 
In Bootstrap 4 theme we are trying to test each component individually and combined as well.

Reply:
cc @jevakallio